### PR TITLE
Implement Capabilities, Get, Set, Subscribe RPCs and IOS XR convenience functions

### PR DESCRIPTION
**Pull request by remingtonc**
All RPCs implemented for usage with Python. Original pull had convenience functions directly implemented to Client object but there appears to be some implementation nuance that is being figured out. Scoping convenience functions on a per-OS basis until figured out. IOS XR convenience functions implemented as `delete_xpaths`, `get_xpaths`, `set_json`, and `subscribe_xpaths` to ease usage of RPCs.

~~Not sure what to call it yet, hesitant to publish as `gnmi` directly - but don't want to completely call it `cisco_gnmi` as it's not a specific Cisco interface. Might look at donating library to OpenConfig.~~ Due to the Cisco specific nature of some behavior/wrappers will likely rename to `cisco-gnmi-python` or something along those lines.

Not merging until naming is determined.

### CODE REVIEW

**Comment by remingtonc** [comment]
There is some platform-specific behavior which is going to be frustrating to figure out. Closing until determined.

**Comment by remingtonc** [comment]
ping

File: README.md [hunk 1]
```diff
@@ -1,2 +1,72 @@
...
+
+Methods are documented in [`gnmi/xr.py`](gnmi/xr.py).
+
+## gNMI
+gRPC Network Management Interface (gNMI) is a service defining an interface for a network management system to interact with a network element. It may be thought of as akin to NETCONF or other control protocols which define operations and behaviors. The scope of gNMI is relatively simple - it seeks to "[[define](https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-specification.md)] a gRPC-based protocol for the modification and retrieval of configuration from a target device, as well as the control and generation of telemetry streams from a target device to a data collection system. The intention is that a single gRPC service definition can cover both configuration and telemetry - allowing a single implementation on the target, as well as a single NMS element to interact with the device via telemetry and configuration RPCs".
```
Comment (ziausyed) [inline README.md]: Do you want to specify what "NMS" is?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,72 @@
 # gnmi-python
-gNMI Python connectivity library. 
+[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
+[![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/ambv/black)
+
+This library wraps gNMI functionality to ease usage with Cisco implementations in Python programs. Derived from [openconfig/gnmi](https://github.com/openconfig/gnmi/proto/gnmi/gnmi.proto).
```
Comment (ziausyed) [inline README.md]: I don't think this link works..

File: README.md [hunk 3]
```diff
@@ -1,2 +1,72 @@
...
+
+## Usage
+```bash
+# TBD
+#pip install gnmi
+```
+
+This library covers the gNMI defined `capabilities`, `get`, `set`, and `subscribe` RPCs, and helper clients provide OS-specific recommendations. As commonalities and differences are identified this library will be refactored as necessary.
+
+It is *highly* recommended that users of the library learn [Google Protocol Buffers](https://developers.google.com/protocol-buffers/) syntax to significantly ease usage. Understanding how to read Protocol Buffers, and reference the `gnmi.proto`, will be immensely useful for utilizing gNMI and any other gRPC interface.
```
Comment (ziausyed) [inline README.md]: change "reference" to "referencing"

File: README.md (hunk 1 above)
Comment (ziausyed) [inline README.md]: The word "define" has an extra set of [] around it, did you intend for that?

File: README.md [hunk 4]
```diff
@@ -1,2 +1,72 @@
...
+```
+
+## Licensing
+`gnmi-python` is licensed as [Apache License, Version 2.0](LICENSE).
+
+## Issues
+Open an issue :)
+
+## Related Projects
+1. [openconfig/gnmi](https://github.com/openconfig-gnmi)
```
Comment (ziausyed) [inline README.md]: This link doesn't work for me

File: README.md (hunk 3 above)
Comment (remingtonc) [reply README.md]: Ooh we read it slightly differently. I was intending to keep this meaning separate. "referencing the `gnmi.proto`" in the context of the sentence would apply to "any other gRPC interface" which is untrue. The broader skill of reading protocol buffers applies to gNMI and any other gRPC interface, but referencing `gnmi.proto` does not.

File: README.md (hunk 1 above)
Comment (remingtonc) [reply README.md]: Was indeed the intention. Quoting the spec, wasn't the exact word utilized.

File: Pipfile [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+name = "pypi"
+
+[dev-packages]
+grpcio-tools = "*"
+googleapis-common-protos = "*"
+pylint = "*"
+black = "==19.3b0"
+
+[packages]
+grpcio = "*"
```
Comment (skkumaravel) [inline Pipfile:16]: Add a min version

File: README.md [hunk 6]
```diff
@@ -1,2 +1,72 @@
 # gnmi-python
-gNMI Python connectivity library. 
+[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
+[![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/ambv/black)
+
+This library wraps gNMI functionality to ease usage with Cisco implementations in Python programs. Derived from [openconfig/gnmi](https://github.com/openconfig/gnmi/blob/master/proto/gnmi/gnmi.proto).
+
+## Usage
+```bash
+# TBD
```
Comment (skkumaravel) [inline README.md]: TBD?

**Review by ziausyed — APPROVED** [review]
nice!

File: hygiene.sh [hunk 7]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+pipenv run black --safe --verbose --exclude proto gnmi
```
Comment (skkumaravel) [inline hygiene.sh]: Add a comment what this file is used for

**Review by skkumaravel — APPROVED** [review]
Looks good small changes. Did not test.

File: README.md (hunk 6 above)
Comment (remingtonc) [reply README.md]: `pip install cisco_gnmi`

File: Pipfile (hunk 5 above)
Comment (remingtonc) [reply Pipfile:16]: I have no opinions on a minimum version and as a library don't want to pin the dependency. Thoughts?

**Pull request merged**